src/shared/dissect-image.c

```diff
@@ -425,10 +425,11 @@ int dissect_image(
 
                         m->encrypted = streq_ptr(fstype, "crypto_LUKS");
 
-                        r = loop_wait_for_partitions_to_appear(fd, d, 0, flags, &e);
-                        if (r < 0)
-                                return r;
-
+                        if (strneq(usage, "filesystem", STRLEN("filesystem"))) {
```

Comment:
`usage` is just a normal string, so `streq` should be used, without the length.

(And STRLEN() should not be used with strneq(). STRLEN() is only for variable declarations to avoid VLA.)

Reply:
Thanks for the review. Somehow I confused `strneq` to mean 'not equal'. This could also explain why some of my tests still failed.

Reply:
Heh. I got so hung up on the STRLEN thing that I didn't notice that the condition is reversed ;)